jarun/buku
# bugfix bukuserver

resolves #543
resolves #545
resolves #547
resolves #548
resolves #553
resolves #554
resolves #559

- pre-commit
  - update
  - use local pylint
- fix import server error on bukuserver.\_\_main\_\_
- isort
  - bukuserver.forms
  - bukuserver.server
  - bukuserver.views
  - tests.test_buku
- don't validate url, let buku do it
- bukuserver
  - use its own search_tag func
  - use fork of flask-reverse-proxy-fix
- pylint fix BookmarkletView too few public method
- tests.test_buku
  - merge test_parse_tags
  - new test_parse_tags_no_args
- simplify test_bookmark_model_view
- flake8 add e203 for black compatilibityh
- bukuserver.views
  - handle unbound choosen_idx CustomAdminIndexView.search
  - BookmarkModelView._apply_filters
    - remove unused flt_name var
    - handle no self._filters
  - BookmarkModelView
    - \_apply_filters
      - black formatted
      - dont manually check scheme validness
    - use buku.parse_tags on create_model and update_model
    - check arg when creating new bookmark record
  - use single message on exception and error flash
  - black formatted
  - dont use favicon when there is no netloc
